scripts/get-sources

```diff
@@ -43,104 +50,125 @@ source "$BUILDER_DIR/scripts/common"
 
 url_var="GIT_URL_${COMPONENT//-/_}"
 
-if [ -n "$GIT_URL" ]; then
+if [ -n "${GIT_URL-}" ]; then
     GIT_URL="$GIT_URL"
-elif [ -n "${!url_var}" ]; then
+elif [ -n "${!url_var-}" ]; then
     GIT_URL="${!url_var}"
 else
     GIT_URL=$GIT_BASEURL/$GIT_PREFIX$COMPONENT$GIT_SUFFIX
 fi
 
 # Override GIT_URL with GIT_REMOTE if given
-[ -n "$GIT_REMOTE" ] && GIT_URL=$GIT_REMOTE
+[ -n "${GIT_REMOTE-}" ] && GIT_URL=$GIT_REMOTE
 
 branch_var="BRANCH_${COMPONENT//-/_}"
 
-if [ -n "${!branch_var}" ]; then
+if [ -n "${!branch_var-}" ]; then
     BRANCH="${!branch_var}"
 fi
 
+: "${IGNORE_MISSING=0}"
+
 GIT_OPTIONS=
 if [ "$GIT_CLONE_FAST" = "1" ]; then
     GIT_OPTIONS+="--depth=1"
 fi
 
-fresh_clone=0
-if [ "$REPO" == "." ] || [ -d "$REPO" ] && [ "$CLEAN" != '1' ]; then
-    cd $REPO
-    if [ "$(git rev-parse --is-shallow-repository)" == "true" ] && [ "$GIT_CLONE_FAST" != "1" ]; then
+if ! [[ "$BRANCH" =~ ^[A-Za-z][A-Za-z0-9._-]+$ ]]; then
```

Comment:
@DemiMarie this reject branches with `/` (which is a valid char in a branch name). Is it intentional, or accidental omission?

Reply:
Intentional, in that a branch name is often used as a path, but I can undo it.

Reply:
Do we do it anywhere?

Reply:
Not that I am aware of, but I have not thoroughly audited the code either.